code/modules/antagonists/changeling/powers/summon_spiders.dm

```diff
@@ -1,18 +1,191 @@
+	. = ..()
+	if(user.UID() != owner_UID)
+		return
+	spider_order(user)
+
+/mob/living/simple_animal/hostile/poison/giant_spider/hunter/infestation_spider/proc/spider_order(mob/user)
+	enemies = list()
+	switch(current_order)
+		if(IDLE_AGGRESSIVE)
+			to_chat(user, "<span class='notice'>We order the giant spider to follow us but attack anyone on sight.</span>")
+			current_order = FOLLOW_AGGRESSIVE
+			handle_automated_movement()
+		if(FOLLOW_AGGRESSIVE)
+			to_chat(user, "<span class='notice'>We order the giant spider to follow us and to remain calm, only attacking if it is attacked.</span>")
+			current_order = FOLLOW_RETALIATE
+			handle_automated_movement()
+		if(FOLLOW_RETALIATE)
+			to_chat(user, "<span class='notice'>We order the giant spider to remain idle and calm, only attacking if it is attacked.</span>")
+			current_order = IDLE_RETALIATE
+		if(IDLE_RETALIATE)
+			to_chat(user, "<span class='notice'>We order the giant spider to remain idle, but ready to attack anyone on sight.</span>")
+			current_order = IDLE_AGGRESSIVE
+	handle_automated_movement()
+
+/mob/living/simple_animal/hostile/poison/giant_spider/hunter/infestation_spider/handle_automated_movement() //Hacky and ugly.
+	. = ..()
+	var/list/around = view(src, vision_range)
+	switch(current_order)
+		if(IDLE_AGGRESSIVE)
+			Find_Enemies(around)
+			walk(src, 0)
+		if(FOLLOW_AGGRESSIVE)
+			Find_Enemies(around)
+			for(var/mob/living/carbon/C in around)
+				if(!faction_check_mob(C))
+					continue
+				if(Adjacent(C))
+					return TRUE
+				Goto(C, 0.5 SECONDS, 1)
+		if(FOLLOW_RETALIATE)
+			for(var/mob/living/carbon/C in around)
+				if(!faction_check_mob(C))
+					continue
+				if(Adjacent(C))
+					return TRUE
+				Goto(C, 0.5 SECONDS, 1)
+		if(IDLE_RETALIATE)
+			walk(src, 0)
+
+	for(var/mob/living/simple_animal/hostile/poison/giant_spider/hunter/infestation_spider/H in around)
+		if(faction_check_mob(H) && !attack_same && !H.attack_same)
+			H.enemies |= enemies
 
-	SSblackbox.record_feedback("nested tally", "changeling_powers", 1, list("[name]"))
 	return TRUE
+
+// Bellow is the way the spiders react and retaliate when in an idle/aggresive mode.
+
+/mob/living/simple_animal/hostile/poison/giant_spider/hunter/infestation_spider/ListTargets()
+	if(!length(enemies))
+		return list()
+	var/list/see = ..()
+	see &= enemies // Remove all entries that aren't in enemies
+	return see
+
+/mob/living/simple_animal/hostile/poison/giant_spider/hunter/infestation_spider/proc/Find_Enemies(around)
+	enemies = list() // Reset enemies list, only focus on the ones around you, spiders don't have grudges
+	for(var/mob/living/A in around)
+		if(A == src)
+			continue
+		if(!isliving(A))
+			continue
+		var/mob/living/M = A
+		if(!faction_check_mob(M))
+			enemies |= M
+
+/mob/living/simple_animal/hostile/poison/giant_spider/hunter/infestation_spider/attackby(obj/item/W, mob/user, params)
+	..()
```

Comment:
You probably want to do . = ..() for most of these, otherwise it might do weird things to the attack chain.

Reply:
Handled!